helper/semvercon/semvercon.go

```diff
@@ -0,0 +1,147 @@
+// semvercon is a Semver Constraints package copied from
+// github.com/hashicorp/go-version @ 2046c9d0f0b03c779670f5186a2a4b2c85493a71
+//
+// Unlike Constraints in go-version, Semver constraints use Semver 2.0 ordering
+// rules and only accept properly formatted Semver versions.
+package semvercon
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+)
+
+// SemverConstraint represents a single constraint for a version, such as ">=
+// 1.0".
+type SemverConstraint struct {
+	f        constraintFunc
+	check    *version.Version
+	original string
+}
+
+// SemverConstraints is a slice of constraints. We make a custom type so that
+// we can add methods to it.
+type SemverConstraints []*SemverConstraint
+
+type constraintFunc func(v, c *version.Version) bool
+
+var constraintOperators map[string]constraintFunc
+
+var constraintRegexp *regexp.Regexp
+
+func init() {
+	constraintOperators = map[string]constraintFunc{
+		"":   constraintEqual,
+		"=":  constraintEqual,
+		"!=": constraintNotEqual,
+		">":  constraintGreaterThan,
+		"<":  constraintLessThan,
+		">=": constraintGreaterThanEqual,
+		"<=": constraintLessThanEqual,
+	}
+
+	ops := make([]string, 0, len(constraintOperators))
+	for k := range constraintOperators {
+		ops = append(ops, regexp.QuoteMeta(k))
+	}
+
+	constraintRegexp = regexp.MustCompile(fmt.Sprintf(
+		`^\s*(%s)\s*(%s)\s*$`,
+		strings.Join(ops, "|"),
+		version.SemverRegexpRaw))
+}
+
+// NewSemverConstraint will parse one or more constraints from the given
+// constraint string. The string must be a comma-separated list of constraints.
+func NewSemverConstraint(v string) (SemverConstraints, error) {
```

Comment:
Could rename to `NewConstraint` and change the types to `Constraint` and `Constraints` given you are in the `semver` package.